tools/flang1/flang1exe/lowersym.c

```diff
@@ -1239,14 +1239,25 @@ replace_fval_in_params(SPTR entry, SPTR entrysame)
   SPTR fval, fvalsame, newdsc, newdscsame, newarg, newargsame;
   int params, narg, i;
 
+  narg = PARAMCTG(entry);
+  if (narg == 0)
+    return;
+  params = DPDSCG(entry);
   fval = FVALG(entry);
   fvalsame = FVALG(entrysame);
   newdsc = NEWDSCG(fval);
   newarg = NEWARGG(fval);
+  /* If the return variable is an adjustable length character without POINTER
+   * attribute, its NEWARG keeps SPTR_NULL(see init_newargs in bblock.c), and
+   * itself will be put into dummy parameters(see newargs_for_entry in
+   * dpm_out.c), so we match and replace the return variable directly.
+   */
+  if (newarg == 0)
```

Comment:
Instead of 0, I suggest using `SPTR_NULL`.

Reply:
> Instead of 0, I suggest using `SPTR_NULL`.

Fixed.